src/main.rs

```diff
@@ -696,8 +696,14 @@ fn start_vmm(cmd_arguments: ArgMatches) -> Result<Option<String>, Error> {
     .map_err(Error::StartVmmThread)?;
 
     let r: Result<(), Error> = (|| {
-        let payload_present =
-            cmd_arguments.contains_id("kernel") || cmd_arguments.contains_id("firmware");
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "igvm")] {
+                let payload_present = cmd_arguments.contains_id("kernel") || cmd_arguments.contains_id("firmware") || cmd_arguments.contains_id("igvm");
+            } else {
+                let payload_present =
+                    cmd_arguments.contains_id("kernel") || cmd_arguments.contains_id("firmware");
+            }
+        }
```

Comment:
I believe it would more concise and read easier if you simply use `#[cfg(not(feature = "igvm"))]` and `#[cfg(feature = "igvm")]`. My understanding is that `cfg_if::cfg_if` would be useful for more complicated cases.